frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts

```diff
@@ -70,8 +72,17 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
             (taxonomicFilterLogicKey) => taxonomicFilterLogicKey,
         ],
         taxonomicGroups: [
-            (selectors) => [selectors.currentTeamId],
-            (teamId): TaxonomicFilterGroup[] => [
+            (selectors) => [selectors.currentTeamId, selectors.groupTypes],
+            (teamId, groupTypes): TaxonomicFilterGroup[] => [
+                ...(groupTypes.length > 0
+                    ? groupTypes.map((gt) => ({
+                          name: capitalizeFirstLetter(gt.group_type),
+                          type: gt.group_type,
+                          endpoint: `api/projects/${teamId}/groups`,
+                          getName: (group: GroupType) => group.group_type,
+                          getValue: (group: GroupType) => group.group_type_index,
+                      }))
+                    : []),
```

Comment:
The ternary is not needed - if you have an empty array then mapping over an empty array yields an empty array.